server/channels/app/imaging/preview.go

```diff
@@ -34,17 +34,13 @@ func GenerateThumbnail(img image.Image, width, height int) image.Image {
 	if h > height || w > width {
 		ratio := float64(h) / float64(w)
 		if ratio < expectedRatio {
-			// we pre-calculate the thumbnail's width to make sure we are not upscaling.
```

Comment:
Could we reformulate but otherwise leave the comment? I think it was useful to understand the reason behind the extra complexities in this function.

Reply:
Added a comment explaining the logic.

Reply:
@turretkeeper Why do we no longer need to check if we're upscaling or not?

Reply:
I am not sure what this comment was referring to. If you disregard the comment and integer rounding, the logic should be exactly the same between the two versions - see the PR description for an explanation.